### DISCUSSION TITLE
What do you think about this idea?

### DISCUSSION THREAD

**Discussion by frederikhors** [Ernxst/svelte-query-params#7] [Ideas]
### Clear and concise description of the problem

Can I ask you what do you think about https://github.com/sveltejs/kit/discussions/12392?

### Suggested solution

Sorry if I opened an issue for this but the discussions tab is off.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/Ernxst/svelte-query-params/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Ernxst/svelte-query-params/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://github.com/Ernxst/svelte-query-params/blob/main/docs).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by Ernxst** [comment]
Since it's SvelteKit specific, you could try converting it into a custom SvelteKit adapter that would conform to the [Adapter API](https://github.com/Ernxst/svelte-query-params/blob/main/packages/core/src/lib/adapters/types.ts). The problem you have in your example is it uses `window` which isn't available on the server. This is fine for your use case (you can stub out the server-side methods) - see the official SvelteKit [adapter](https://github.com/Ernxst/svelte-query-params/blob/e544a88110f38627166cadb823fd2afbe22a7d64/packages/core/src/lib/adapters/sveltekit.ts).

**Reply by Ernxst** [reply]
Saying that, you could also try the [`dom`](https://github.com/Ernxst/svelte-query-params/tree/main?tab=readme-ov-file#dom) adapter, which is used by default if you don't specify an adapter; this adapter does nothing on the server.

**Reply by frederikhors** [reply]
I don't know where to start unfortunately. Would you mind working on it? I think many people will love it!
